### ISSUE TITLE
saveAs inside webcomponent

### ISSUE THREAD

**Issue by BaluVyamajala** [eligrey/FileSaver.js#520]
when saveAs for blob is called, its going to this the empty function with just /* noop */ and not doing anything. I can't seem to figure out why code is going to this spot? 
I am using file-saver on a web component created out of angluar 7 and hosted in another angular application.

`var saveAs = _global.saveAs || (
  // probably in some web worker
  (typeof window !== 'object' || window !== _global)
    ? function saveAs () { /* noop */ }
`

**Comment by BaluVyamajala** [comment]
In case when I run the web component separately, _global.saveAs is resulting as undefined , which is helping to not go this blank saveAs function. However, when I run together with in a hosted application, it is resulting in actual function  and taking me to this blank saveAs. Please advice.

**Comment by Ruffio** [comment]
I have no knowledge of web components but I think that you have to adjust the above lines to point to a shadowDOM somehow.

Have you got it resolved yourself?

**Comment by BaluVyamajala** [comment]
yes, I copied the  js file into my application and made changes in my local.
removed _global.saveAs in the condition.
    var saveAs = typeof window !== 'object' || window !== _global
